pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -255,12 +256,65 @@ func configFromEnv() (cfg Config, ok bool) {
 	return
 }
 
+// Allows Setting up of openstack credentials by reading clouds.yaml
+//
+// Looks for clouds.yaml file in the following paths in order:
+// 1. OS_CLIENT_CONFIG_FILE
+// 2. Current directory.
+// 3. unix-specific user_config_dir (~/.config/openstack/clouds.yaml)
+// 4. unix-specific site_config_dir (/etc/openstack/clouds.yaml)
+func configFromYAML() (cfg Config, err error) {
+	clientOpts := new(clientconfig.ClientOpts)
+	cloud, err := clientconfig.GetCloudFromYAML(clientOpts)
+
+	// Exit as a failure if unable to read clouds.yaml
+	if err != nil {
+		fmt.Println(err)
+		return cfg, err
+	}
+
+	cfg.Global.AuthURL = cloud.AuthInfo.AuthURL
+	cfg.Global.Username = cloud.AuthInfo.Username
+	cfg.Global.Password = cloud.AuthInfo.Password
+	cfg.Global.Region = cloud.RegionName
+	cfg.Global.UserID = cloud.AuthInfo.UserID
+	cfg.Global.TenantID = cloud.AuthInfo.ProjectID
+	cfg.Global.TenantName = cloud.AuthInfo.ProjectName
+	cfg.Global.DomainID = cloud.AuthInfo.UserDomainID
+	cfg.Global.DomainName = cloud.AuthInfo.UserDomainName
+	cfg.Global.CAFile = cloud.CACertFile
+
+	// Retaining support for trusts via environment variables
+	// due to lack of support by GetCloudFromYAML(gohpercloud)
+	cfg.Global.TrustID = os.Getenv("OS_TRUST_ID")
+
+	// Supporting injecting passwords via env variables
+	if cfg.Global.Password == "" {
```

Comment:
We shouldn't need this now that we're using `clientconfig`.

Reply:
mmh, if that's the case, I'd move the Trusts field to the main Config struct. We can still use it from there.